pravega/pravega-benchmark
# Closing Readergroups, writers and adding "-flush" parameter to Benchmark tool

Change log description
Currently the reader groups are getting closed once the read compete; This  pull request ensures that reader groups and writers are closed even when the  user terminates the bench-marking.

Purpose of the change
Fixes #37, #41 , #43 

What the code does
To handle ctrl+c ,  the method to call closure of all readers and writer are invoked.
-flush <N>" parameter is added to the pravega benchmark tool.
where N is number of messages to flush per producer.
This option can be used to flush the message by producers while
measuring write performance.


How to verify it
build the tool and issue the writes with -flush <number of events> to verify the issue #41.
run the benchmark tool with consumers and press Ctrl+C while reading ; and re issue the same command to read the data from the beginning again.

Signed-off-by: Keshava Munegowda keshava.munegowda@dell.com
